llvm/include/llvm/MC/MCDecoder.h

```diff
@@ -58,6 +59,24 @@ uint64_t fieldFromInstruction(const std::bitset<N> &Insn, unsigned StartBit,
   return ((Insn >> StartBit) & Mask).to_ullong();
 }
 
+template <unsigned StartBit, unsigned NumBits, typename T>
+inline std::enable_if_t<std::is_unsigned_v<T>, T> extractBits(T Val) {
+  static_assert(StartBit + NumBits <= std::numeric_limits<T>::digits);
+  return (Val >> StartBit) & maskTrailingOnes<T>(NumBits);
+}
+
+template <unsigned StartBit, unsigned NumBits, size_t N>
+uint64_t extractBits(const std::bitset<N> &Val) {
+  static_assert(StartBit + NumBits <= N);
+  std::bitset<N> Mask = maskTrailingOnes<uint64_t>(NumBits);
+  return ((Val >> StartBit) & Mask).to_ullong();
+}
```

Comment:
is this override being used?

Reply:
Yes, by AMDGPU and RISCV.
The APInt overload is used by M68k.

Reply:
Right, with --specialize-decoders-per-bitwidth we switched AMDGPU and RISCV to use std::bitset for 48 and 96/128 bit insts